### DISCUSSION TITLE
[github] Release changes not incremental compared to previous release

### DISCUSSION THREAD

**Discussion by mthmulders** [jreleaser/jreleaser#846] [Q&A]
I finally found some time to update my project, [MCS](https://github.com/mthmulders/mcs), to JReleaser 1.0.0. Long overdue, I know.... I was surprised with two things, I'll open separate discussions for them. This is 1/2.

After upgrading from JReleaser 0.1.0 to 1.0.0, and pushing a new MCS release, I noticed that the GitHub release page lists the complete history of the project - not just the changes compared to the previous release. Is this intentional? Or did I miss something in the release notes that could have pointed me to this?

For good measure: the release I just rolled (0.2.1) should contain only Maven plugins and dependency updates. No new features whatsoever. So the list in the GitHub release page is definitely too long...

**Comment by aalmiray** [comment]
No, this is not the default behavior. JReleaser should had included all commits from the last tag to the current one. When this does not happen in means that the tool failed to recognize the last tag. 

This could also be caused by a manually created tag whose sort order is higher than the expected value. You can see what's the expected value if you look at the `trace.log` file. If the chosen tag value is wrong then perhaps the current set of tags might be sketchy (`git tag` to check which tags are attached to the repo).

**Reply by mthmulders** [reply]
Unfortunately, I don't have the trace.log anymore, but I [took care to save it](https://github.com/mthmulders/mcs/commit/0d09c87c571632a18a4f757c2504bfdb4fcb7da5) next time an issue might happen.

The current list of tags is

```
early-access
v0.1
v0.1.1
[... skip ...]
v0.1.9
v0.2
v0.2.1
```

I would say it makes sense, but I guess it's hard to tell from here...

**Reply by aalmiray** [reply]
Here's what the log says

```
[DEBUG] looking for tag that matches 'v0.2.1'
[DEBUG] looking for tags that match 'v.*', excluding 'v0.2.1'
[DEBUG] found tag v0.1.18
[DEBUG] sorting commits DESC
```

I think the problem is that `v0.2` does not conform to the detected semver patter (x.y.z) and thus is excluded. That tag should had been `v0.2.0`.
